store/src/java/com/zimbra/cs/service/account/Auth.java

```diff
@@ -321,12 +327,12 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
                         AppSpecificPasswords appPasswords = TwoFactorAuth.getFactory().getAppSpecificPasswords(acct, acctValuePassedIn);
                         appPasswords.authenticate(password);
                     } else {
-                        prov.authAccount(acct, code, AuthContext.Protocol.soap, authCtxt);
+                        prov.authAccount(acct, code, context.get(SoapEngine.REQUEST_PROTO, AuthContext.Protocol.soap), authCtxt);
```

Comment:
Please change this line to -
prov.authAccount(acct, code, (AuthContext.Protocol)context.get(SoapEngine.REQUEST_PROTO), authCtxt);